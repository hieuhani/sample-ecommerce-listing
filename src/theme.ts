import { DefaultTheme } from 'styled-components'

export const theme: DefaultTheme = {
  colors: {
    reddishOrange: '#f5471e',
    coolGrey: '#8f9598',
    silver: '#c6cace',
    tomato: '#ea341f',
    darkGrey: '#262829',
    paleGrey: '#eef1f3',
    white: '#FFF',
    black30: 'rgba(0, 0, 0, .3)',
    pumpkinOrange: '#f58400',
    deepSkyBlue: '#1e75f6',
  }
}
