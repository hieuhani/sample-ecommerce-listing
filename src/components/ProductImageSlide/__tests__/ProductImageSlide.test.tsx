import React from 'react'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import { render } from '@testing-library/react'
import { ProductImageSlide } from '..'
import { theme } from '../../../theme'

const props = {
  image: {
    url: 'https://devdocs.magento.com/assets/i/magento-logo.svg',
    priority: 1,
  },
}

describe('ProductImageSlide', () => {
  test('ProductImageSlide renders correctly', () => {
    const tree = renderer.create(
      <ThemeProvider theme={theme}>
        <ProductImageSlide {...props} />
      </ThemeProvider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('AddToCartButton displays Add To Cart text when totalValue === 0', () => {
    const testProps = props
    const { container } = render(
      <ThemeProvider theme={theme}>
        <ProductImageSlide {...testProps} />
      </ThemeProvider>
    )
    expect(container.querySelector('img')).toHaveAttribute('src', props.image.url)
  })
})
