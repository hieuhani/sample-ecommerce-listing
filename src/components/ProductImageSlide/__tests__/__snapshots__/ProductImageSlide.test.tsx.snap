// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImageSlide ProductImageSlide renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #FFF;
}

.c1 {
  max-width: 330pt;
}

.c1 img {
  width: 100%;
}

<div
  className="c0 swiper-slide"
>
  <div
    className="c1"
  >
    <img
      src="https://devdocs.magento.com/assets/i/magento-logo.svg"
    />
  </div>
</div>
`;
